nvdaHelper/UIARemote/lowLevel.cpp

```diff
@@ -0,0 +1,191 @@
+		return E_FAIL;
+	}
+	AutomationTextRange textRange {nullptr};
+	winrt::copy_from_abi(textRange, arg_pTextRange);
+	if(!textRange) {
+		return E_FAIL;
+	}
+	operation.ImportTextRange(AutomationRemoteOperationOperandId{arg_registerID}, textRange);
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOp_addToResults(void* arg_pRemoteOp, int arg_registerID) {
+	CoreAutomationRemoteOperation operation {nullptr};
+	winrt::copy_from_abi(operation, arg_pRemoteOp);
+	if(!operation) {
+		return E_FAIL;
+	}
+	operation.AddToResults(AutomationRemoteOperationOperandId{arg_registerID});
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOp_isOpcodeSupported(void* arg_pRemoteOp, uint32_t arg_opcode, bool* pIsSupported) {
+	CoreAutomationRemoteOperation operation {nullptr};
+	winrt::copy_from_abi(operation, arg_pRemoteOp);
+	if(!operation) {
+		return E_FAIL;
+	}
+	*pIsSupported = operation.IsOpcodeSupported(arg_opcode);
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOp_execute(void* arg_pRemoteOp, uint8_t arg_byteCodeBuffer[], int arg_byteCodeBufferLength, void** ppResults) {
+	CoreAutomationRemoteOperation operation {nullptr};
+	winrt::copy_from_abi(operation, arg_pRemoteOp);
+	if(!operation) {
+		return E_FAIL;
+	}
+	auto results = operation.Execute(winrt::array_view<uint8_t>(arg_byteCodeBuffer, arg_byteCodeBufferLength));
+	*ppResults = winrt::detach_abi(results);
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOpResult_getErrorLocation(void* arg_pResults, int* pErrorLocation) {
+	AutomationRemoteOperationResult results {nullptr};
+	winrt::copy_from_abi(results, arg_pResults);
+	if(!results) {
+		return E_INVALIDARG;
+	}
+	*pErrorLocation = results.ErrorLocation();
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOpResult_getExtendedError(void* arg_pResults, HRESULT* pExtendedError) {
+	AutomationRemoteOperationResult results {nullptr};
+	winrt::copy_from_abi(results, arg_pResults);
+	if(!results) {
+		return E_INVALIDARG;
+	}
+	*pExtendedError = results.ExtendedError();
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOpResult_getStatus(void* arg_pResults, int* pStatus) {
+	AutomationRemoteOperationResult results {nullptr};
+	winrt::copy_from_abi(results, arg_pResults);
+	if(!results) {
+		return E_INVALIDARG;
+	}
+	auto status = results.Status();
+	*pStatus = static_cast<int>(status);
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT __stdcall remoteOpResult_hasOperand(void* arg_pResults, int arg_registerID, bool* pHasOperand) {
+	AutomationRemoteOperationResult results {nullptr};
+	winrt::copy_from_abi(results, arg_pResults);
+	if(!results) {
+		return E_INVALIDARG;
+	}
+	*pHasOperand = results.HasOperand(AutomationRemoteOperationOperandId{arg_registerID});
+	return S_OK;
+}
+
+extern "C" __declspec(dllexport) void __stdcall remoteOpResult_free(void* arg_pResults) {
```

Comment:
Perhaps for consistency this function should also return `HRESULT` (always `S_OK`) like all the others in the API?